packages/core-v2/src/semantic-model/concepts/concepts.ts

```diff
@@ -40,28 +40,28 @@ export interface SemanticModelClass extends NamedThing, SemanticModelEntity {
 export interface SemanticModelRelationship extends NamedThing, SemanticModelEntity {
     type: [typeof SEMANTIC_MODEL_RELATIONSHIP];
 
-    ends: SemanticModelRelationshipEnd[]
+    ends: SemanticModelRelationshipEnd[];
 
     // todo: is it attribute or association
 }
 
 export interface SemanticModelRelationshipEnd extends NamedThing {
-    cardinality?: [number, number|null];
+    iri: string | null;
+    cardinality?: [number, number | null];
 
     /** {@link SemanticModelClass} */
-    concept: string;
+    concept: string | null;
```

Comment:
Ok, if the concept is `null` we treat is as the user has not yet decided or it is unknown?

Reply:
Yes, seems cleaner this way. We can set the concept to `null` instead of `""` [here](https://github.com/mff-uk/dataspecer/blame/2d3dd3e2400951cbfb28f4d6b0d929fda5af57b8/packages/core-v2/src/semantic-model/writable-semantic-model-adapter.ts#L116).